probes/sastToolRunsOnAllCommits/def.yml

```diff
@@ -0,0 +1,30 @@
+# Copyright 2023 OpenSSF Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+id: sastToolRunsOnAllCommits
+short: Checks that a SAST tool runs on all commits in the projects CI.
+motivation: >
+  SAST is testing run on source code before the application is run. Using SAST tools can prevent known classes of bugs from being inadvertently introduced in the codebase.
+implementation: >
+  The implementation iterates through the projects commits and checks whether any of the check runs for the commits associated merge request was any of the SAST tools that Scorecard supports.
+outcome:
+  - If the project had no commits merged, the probe returns a finding with OutcomeNotApplicable.
+  - If any of the check runs of an associated merge request of a commit was for a SAST tool, the probe returns one finding with OutcomePositive (1). In addition, the finding will include to values. 1) How many commits were tested by a SAST tool, and 2) How many commits in total were merged.
+  - If none of the check runs of an associated merge request of a commit was for a SAST tool, the probe returns one finding with OutcomeNegative (0). In addition, the finding will include to values. 1) How many commits were tested by a SAST tool, and 2) How many commits in total were merged.
```

Comment:
iiuc, if there are 2 SAST tools installed (SA and SB), SA runs on N PRs and SB runs on M PRs (total PRs = N + M), it returns on positive outcome, correct?

Reply:
I would say almost. 

This part is true: "it returns on positive outcome",

However, this part: "if there are 2 SAST tools installed (SA and SB), SA runs on N PRs and SB runs on M PRs (total PRs = N + M)", is not entirely correct.

If there are 2 SAST tools installed, they can both run on the same PRs, and as such, N+M can be greater than the total sum of all PRs that Scorecard checks.

Reply:
cool, that matches my understanding yes. This is useful when users want to upgrade from one tool to another (during the transition period some PRs will have SA run, while others will have SB run).

I was curious if some users will want to say "SA is run on my PRs", because that's the one tool their organization wants their users to run. In this case, maybe we'd create a new probe for each SAST? Not sure this is the best way. Or maybe a probe "sastUniqueRun", which would return multiple results and a `Value['toolName']=the_name` for each SAST tool?

Wdut?